### DISCUSSION TITLE
using yield in route

### DISCUSSION THREAD

**Discussion by HuiiBuh** [fastapi/fastapi#7529] [Questions] [answered]
### Describe the bug

I am not sure if it is a bug, or a feature request. Perhaps using yield in a route is not intended by fastapi.

### To Reproduce
1. Create a file with:

```Python
@app.get("/")
async def root():
    print('hello')
    yield JSONResponse({'some': 'text'})
    print('world')

```

2. Open the browser and call the endpoint `/`.

### Expected behaviour

The {'some': 'text'} as json. 

### Screenshots
But this is what I get. 

![image](https://user-images.githubusercontent.com/46561026/79232076-5752d900-7e67-11ea-8ffe-fbcb9012ba6d.png)

### Environment
- Linux Mint 19
- Fastapi 0.54.1

**Comment by michalgolda** [comment]
As for as i know FastAPI is not support "yield" response. Use "return" instead of "yield". 😁

**Comment by Dustyposa** [comment]
If you want to do something after send a response.
You can try background task

**Comment by HuiiBuh** [comment]
@Maklero Well there is a reason I wanted to use yield and not return ;D

@Dustyposa Thanks I will give it a try

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

Thanks for reporting back and closing the issue @HuiiBuh  :+1:

**Comment by HuiiBuh** [comment]
@tiangolo Just out of interest. Do you consider supporting yield as response?

**Comment by tiangolo** [comment, marked as answer]
@HuiiBuh nope, it wouldn't really make sense, it would add another layer of complexity that wouldn't help much. And it would make the code look as if it was doing something different than it does.

As @Dustyposa says, that would probably belong to BackgroundTasks, it would work better there :rocket: https://fastapi.tiangolo.com/tutorial/background-tasks/
